sorc/lsm_routines.fd/noah.fd/bulk_snow_module.f90

```diff
@@ -0,0 +1,57 @@
+!> @file
+!> @brief Routines to make DA updates to a bulk (single layer) snow model
+!! such as that in Noah.
+!> @author Clara Draper
+
+module bulk_snow_module
+
+ implicit none
+
+ private
+
+ public calc_density
+
+contains
+
+!> This subroutine calculates snow density from forecast fields.
+!! density = SWE/SND where snow present.
+!!         = average from snow forecasts over land, where snow not present
+!! @param[in] lensfc Number of sfc grid cells
+!! @param[in] rank Processor rank
+!! @param[in] landmask Mask for land increments
+!! @param[in] swe Snow Water Equivalent
+!! @param[in] snd Snow Depth
+!! @param[out] density Snow density [-]
+
+ subroutine calc_density(lensfc, landmask, swe, snd, rank, density)
+
+       implicit none
+
+       integer, intent(in) :: lensfc, rank
+       integer, intent(in) :: landmask(lensfc)
+       real, intent(in)    :: swe(lensfc), snd(lensfc)
+       real, intent(out)   :: density(lensfc)
+
+       real :: dens_mean
+
+        ! density = swe/snd
+        density = swe/snd
```

Comment:
I think they called the "calc_density" subroutine where the snow present.

Reply:
We don't use any of these values is there's no snow, but "no snow" is based on SWE, so it could be possible if we somehow wound up with 0.0 snow depth, and non-zero SWE. I replaced that code with a loop, checking on the SND value, just to make sure.

Reply:
> I will hold off merging until @ClaraDraper-NOAA addresses the latest comments.

@barlage @GeorgeGayno-NOAA Just pushed responses to Mike's comments. I'll keep an eye on the automatic checks ( there's a small code change to the snow density calculation, but it passed the snow increment reg_test)